source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -58,7 +58,10 @@ DnsCacheImpl::DnsCacheImpl(
     // cache to load an entry. Further if this particular resolution fails all the is lost is the
```

Comment:
(I'm trying to comment on line 56)

Wow, is this the correct type for hostname?

repeated config.core.v3.[SocketAddress](https://source.corp.google.com/piper///depot/google3/third_party/envoy/src/api/envoy/config/core/v3/address.proto;rcl=580630422;l=54) [preresolve_hostnames](https://source.corp.google.com/piper///depot/google3/third_party/envoy/src/api/envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto;bpv=1;bpt=1;rcl=580630422;l=134?gsn=preresolve_hostnames&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fgoogle3%3Flang%3Dprotobuf%3Fpath%3Dthird_party%2Fenvoy%2Fsrc%2Fapi%2Fenvoy%2Fextensions%2Fcommon%2Fdynamic_forward_proxy%2Fv3%2Fdns_cache.proto%234.1.2.11) = 10;

Hostname is a bizarre term for this as it seems to be a protocol (tcp/udp), an IP address, a port, and a resolver name. Am I reading that correctly? *bizarre*

Reply:
I was also surprised SocketAddress was used for preresolved hostnames config, and doesn't seem to be the right structure to represent a DNS cache entry.  Changing it is a good idea but probably outside the scope of the PR.  I do share your concern though..